server/alembic/versions/8bfeffa2265c_convert_detractor_escalations_to_jsonb.py

```
@@ -0,0 +1,115 @@
+        error_details = "\n".join([
+            f"  Call ID {r['call_id']}: {r['error']}"
+            for r in invalid_rows[:10]  # Show first 10 errors
+        ])
+        total_errors = len(invalid_rows)
+        raise Exception(
+            f"\n❌ Migration ABORTED: Found {total_errors} row(s) with invalid JSON.\n"
+            f"No changes were made to the database.\n\n"
+            f"Invalid rows:\n{error_details}\n"
+            f"{f'...and {total_errors - 10} more' if total_errors > 10 else ''}\n\n"
+            f"Please fix the invalid JSON data before running this migration."
+        )
```

Ruff (0.14.14)

[warning] Create your own exception

(TRY002)